BUILD.md

```diff
@@ -0,0 +1,170 @@
+# Build FreeOrion
+
+
+## Hardware and OS Requirements
+
+FreeOrion will compile on most recent Windows, Mac OS X and Linux operating
+systems. Other operating systems have reported to work, but support is not
+actively maintained by the FreeOrion developers. FreeOrion is developed for x86
+compatible processor architectures, other architectures haven't been tested.
+
+The FreeOrion source code makes heavy use of templates and requires much memory
+to compile ; 4GB RAM or more is recommended. To build FreeOrion, expect up to
```

Comment:
That information probably should have been updated on the wiki long ago. Building FO on a system with merely 4GB RAM has become extremely difficult with the reimplementation of the parser (switching to boost spirit), which happened around the time when I joined the project (autumn 2011). I'm not sure if you can build current versions of FO even with 6GB RAM anymore.

To avoid frustration and make things easier for us, I'd recommend to raise the minimum RAM requirement for building FO to 8GB.

Reply:
> That information probably should have been updated on the wiki long ago. Building FO on a system with merely 4GB RAM has become extremely difficult with the reimplementation of the parser (switching to boost spirit), which happened around the time when I joined the project (autumn 2011). I'm not sure if you can build current versions of FO even with 6GB RAM anymore.

YMMV, I'm building on a 6GB memory 6 core machine with almost full utilization via `-j 5` and it works just fine.  I would rather drop a specific number or remove the section completely.

Reply:
I wonder why this comment is now marked as outdated. I haven't touched that line and I would still like to discuss this.

Reply:
> I'm building on a 6GB memory 6 core machine with almost full utilization via -j 5 and it works just fine.

Yeah, on your Linux system (which, besides being Linux, I guess is excellently maintained). Don't know how Windows 8 and higher do, but in my experience Windows isn't exactly the most memory efficient OS around.

But what I *do* know is that, with only 4GB RAM, I never got FO to successfully build on Windows and only with difficulties by employing some tricks on OSX. So 6GB is probably the minimum, at least on these platforms.

> I would rather drop the specific memory size statement or remove the section completely.

Why? RAM is a significant limiting factor, exactly the thing I'd expect to see in sections dealing with hardware requirements. When it comes to building FO, it's actually the only one which you can possibly not have enough of.

Reply:
Fixed.